src/ConservationChecker.jl

```diff
@@ -16,193 +16,184 @@ using UnPack
-        Δz_top = round(FT(0.5) * (z[end, 1, 1, 1, 1] - z[end - 1, 1, 1, 1, 1]))
-        n_faces = length(z[:, 1, 1, 1, 1])
-
-        LWd_TOA = Fields.level(
-            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_lw_flux_dn), face_space),
-            n_faces - half,
-        )
-        LWu_TOA = Fields.level(
-            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_lw_flux_up), face_space),
-            n_faces - half,
-        )
-        SWd_TOA = Fields.level(
-            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_sw_flux_dn), face_space),
-            n_faces - half,
-        )
-        SWu_TOA = Fields.level(
-            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_sw_flux_up), face_space),
-            n_faces - half,
-        )
-
-        coupler_sim.fields.F_radiative_TOA .-=
-            swap_space!(zeros(boundary_space), LWd_TOA .+ SWd_TOA .- LWu_TOA .- SWu_TOA) .* coupler_sim.Δt_cpl
-        radiation_sources_accum = sum(coupler_sim.fields.F_radiative_TOA) # accumulated radiation sources + sinks [J]
-        push!(cc.toa_net_source, radiation_sources_accum)
-    else
-        push!(cc.toa_net_source, FT(0))
-    end
+    ccs = cc.sums
+    @unpack model_sims = coupler_sim
 
-    # save atmos
-    push!(cc.ρe_tot_atmos, sum(atmos_sim.integrator.u.c.ρe_tot))
+    boundary_space = coupler_sim.boundary_space # thin shell approx (boundary_space[z=0] = boundary_space[z_top])
 
-    # save land
-    if land_sim !== nothing
-        e_per_area_land = zeros(axes(land_sim.integrator.u.bucket.W))
-        get_land_energy(land_sim, e_per_area_land)
-        push!(cc.ρe_tot_land, sum(e_per_area_land .* surface_fractions.land))
-    else
-        push!(cc.ρe_tot_land, FT(0))
-    end
+    FT = eltype(coupler_sim.fields[1])
+
+    total = 0
+
+    # save surfaces
+    for sim in model_sims
+        sim_name = Symbol(Interfacer.name(sim))
+        if sim isa Interfacer.AtmosModelSimulation
+            # save radiation source
+            parent(coupler_sim.fields.F_radiative_TOA) .= parent(Interfacer.get_field(sim, Val(:F_radiative_TOA)))
+
+            if isempty(ccs.toa_net_source)
+                radiation_sources_accum = sum(coupler_sim.fields.F_radiative_TOA .* coupler_sim.Δt_cpl) # ∫ J / m^2 dA
+            else
+                radiation_sources_accum =
+                    sum(coupler_sim.fields.F_radiative_TOA .* coupler_sim.Δt_cpl) .+ ccs.toa_net_source[end] # ∫ J / m^2 dA
+            end
+            push!(ccs.toa_net_source, radiation_sources_accum)
+
+            # save atmos
+            previous = getproperty(ccs, sim_name)
+            current = sum(Interfacer.get_field(sim, Val(:energy))) # # ∫ J / m^3 dV
+
+            push!(previous, current)
+            total += current + radiation_sources_accum
+
+        elseif sim isa Interfacer.SurfaceModelSimulation || sim isa Interfacer.SurfaceStub
```

Comment:
I believe we have `SurfaceStub <: SurfaceModelSimulation`, so this condition could be `elseif sim <: Interfacer.SurfaceModelSimulation`

Reply:
Ahh well spotted! You're right, `Interfacer.SurfaceModelSimulation` should be enough.